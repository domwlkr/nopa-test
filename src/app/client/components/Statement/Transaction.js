import React from 'react';

const Transaction = (props) => {

}

export default Transaction;