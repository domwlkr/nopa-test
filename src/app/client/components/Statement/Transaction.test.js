import React from 'react';
import Transaction from './Transaction';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const singleTransaction = {'id': 1, 'dateStr': 'Now', 'beneficary': 'PAYPAL ZARA', 'value': '- £35.98'};

test('Renders correctly', () => {
  const transaction = renderer.create(
    <Transaction transaction={singleTransaction} />
  ).toJSON();

  expect(transaction).toMatchSnapshot();
});

test('Displays the correct beneficary', () => {
  const transaction = shallow(
    <Transaction transaction={singleTransaction} />
  );

  expect(transaction.instance().props.transaction.beneficary).toEqual('PAYPAL ZARA');
});

test('Displays the correct value', () => {
  const transaction = shallow(
    <Transaction transaction={singleTransaction} />
  );

  expect(transaction.instance().props.transaction.value).toEqual('- £35.98');
});