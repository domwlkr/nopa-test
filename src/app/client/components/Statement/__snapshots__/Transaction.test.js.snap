// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="transaction"
>
  <span>
    PAYPAL ZARA
  </span>
  <span>
    - Â£35.98
  </span>
</div>
`;
